pkgs/top-level/dhall-packages.nix

```diff
@@ -8,20 +8,40 @@ let
     let
       callPackage = newScope self;
 
+      prefer = version: path:
+        let
+          packages = callPackage path { };
+
+        in
+          packages."${version}".overrideAttrs (_: {
+              passthru = packages;
+            }
+          );
+
       buildDhallPackage =
         callPackage ../development/interpreters/dhall/build-dhall-package.nix { };
 
+      buildDhallGitHubPackage =
+        callPackage ../development/interpreters/dhall/build-dhall-github-package.nix { };
```

Comment:
Does it make sense to specialize this to github in nixpkgs? The dhall builder can just assume that `pkgs.fetchFromGitHub` exists (it’s a stable interface of nixpkgs) and then put the result into `src`, which makes it easily extendable to other source locations.

Reply:
@Profpatsch: Yeah, it is possible to do that.  Conceptually, that is what the lower-level `buildDhallPackage` utility already does, it just takes a `src` as input and `buildDhall{GitHub,Directory}Package` are just convenient wrappers around the `buildDhallPackage`.

The main reason I created these intermediate utilities was to keep the code generated by `dhall-to-nixpkgs` as small as possible, otherwise it would essentially be inlining the code that you see in `build-dhall-github-package.nix` into every generated GitHub derivation (including the `mkOverridable` and the `removeAttrs` boilerplate)